### ISSUE TITLE
MutualInformationAffine example failing

### ISSUE THREAD

**Issue by jhlegarreta** [InsightSoftwareConsortium/ITKExamples#178]
**Description**

CIs are reporting a failure for the `MutualInformationAffineTest`:
https://github.com/InsightSoftwareConsortium/ITKExamples/runs/910067190?check_suite_focus=true

or here
https://github.com/InsightSoftwareConsortium/ITKExamples/runs/910067205?check_suite_focus=true

The log of the former reads:
```
test 136
       Start 136: MutualInformationAffineTest
136: Test command: /Users/runner/work/ITKExamples/build/bin/MutualInformationAffine
136: Test timeout computed to be: 1500
136: ExceptionObject caught !
136: 
136: itk::ExceptionObject (0x7fbaa3f2eb60)
136: Location: "unknown" 
136: File: /Users/runner/work/ITKExamples/ITK/Modules/Registration/Common/include/itkMutualInformationImageToImageMetric.hxx
136: Line: 193
136: Description: itk::ERROR: itk::ERROR: MutualInformationImageToImageMetric(0x7fbaa3d05850): All the sampled point mapped to outside of the moving image
136: 
136: 
19/464 Test #136: MutualInformationAffineTest ...................................................***Failed    0.19 sec
``` 

The output image as displayed in the examples site does not seem satisfactory:
https://itk.org/ITKExamples/src/Core/Transform/MutualInformationAffine/Documentation.html

Also, it looks like the output is not really being compared to the baseline; the only reference to a baseline comparison concerning that example shows that one of the input examples is being tested:
```
Start 126: MutualInformationAffineTestBaselineComparison
126: Test command: /Users/runner/work/ITKExamples/build/bin/ImageCompareCommand "--test-image" "/Users/runner/work/ITKExamples/build/src/Core/Transform/MutualInformationAffine/fixed.png" "--baseline-image" "/Users/runner/work/ITKExamples/build/src/Core/Transform/MutualInformationAffine/fixed.png"
2020-07-25T17:08:13.0296100Z 126: Test timeout computed to be: 1500
126: <DartMeasurementFile name="BaselineImage" type="image/png">/Users/runner/work/ITKExamples/build/src/Core/Transform/MutualInformationAffine/fixed.png.base.png</DartMeasurementFile>
126: <DartMeasurementFile name="TestImage" type="image/png">/Users/runner/work/ITKExamples/build/src/Core/Transform/MutualInformationAffine/fixed.png.test.png</DartMeasurementFile>
126: 0
250/464 Test #126: MutualInformationAffineTestBaselineComparison .................................   Passed
```

Have not investigated this thoroughly, but it may have to do with the way in which baseline names are specified in the `CMakeLists.txt`:
https://github.com/InsightSoftwareConsortium/ITKExamples/blob/master/src/Core/Transform/CMakeLists.txt

**Versions**
Current `master` (commit de13e14)

**Comment by jhlegarreta** [comment]
Related to the way in which baselines are specified, wondering if there was a transition towards a given syntax at some point that has not been completed, e.g. another example of specifying the baselines in the tests:
https://github.com/InsightSoftwareConsortium/ITKExamples/blob/master/src/Filtering/ImageGradient/CMakeLists.txt

Maybe a separate issue would be best to keep track of this.

**Comment by thewtex** [comment]
Since this example looks to have come from the ITK Software Guide, I am wondering there is a version in the ITK repository that has been updated?

**Comment by jhlegarreta** [comment]
You are right @thewtex, from the LaTeX-related comments we can conclude that this was an ITK Software Guide example at its origin:
https://github.com/InsightSoftwareConsortium/ITK/commits/master/Examples/RegistrationITKv4/ImageRegistration2.cxx

There are differences when having a close look at both examples. Also, the input images are different. 

And another look at the example, [this line](https://github.com/InsightSoftwareConsortium/ITKExamples/blob/master/src/Core/Transform/MutualInformationAffine/Code.cxx#L175) points exactly to some parameter that produced the reported failure. It looks like the new value does not do a good job either.

So maybe the example requires adjusting the parameters.

Although at some point I guess we would like to have some way to draw more examples from the ITK Software Guide, for now, I would also remove the LaTeX-related comments in here.

**Comment by tbirdso** [comment]
The `MutualInformationAffine` example has been temporarily disabled from being built with ITKExamples as of [PR 197](https://github.com/InsightSoftwareConsortium/ITKExamples/pull/197), but an investigation of example parameters is still needed before we can re-enable the example and resolve the CI failure.

The `MutualInformation` example is worth considering in the investigation as well, as the example results seem to translate the moving image away from the fixed image. The primary difference between these two examples is the type of transform used. [https://itk.org/ITKExamples/src/Registration/Common/MutualInformation/Documentation.html](https://itk.org/ITKExamples/src/Registration/Common/MutualInformation/Documentation.html).
